renderer.cpp

```diff
@@ -10,12 +10,21 @@
 
 #include <iostream>
 
+namespace pong
+{
+
 Renderer& Renderer::GetInstance()
 {
     static Renderer sInstance;
     return sInstance;
 }
 
+void Renderer::Cleanup()
+{
+    GetInstance().mShader.reset();
+    GetInstance().mShader = nullptr;
+}
```

Comment:
Is there a reason this doesn't happen in a destructor?

Reply:
Yeah the issue I was having was that after closing the window, the program would not terminate until manually exited with Ctrl + C. The reason was because after calling glfwTerminate() the shader was still active since it was a static variable. In ~Shader() it cleans up in opengl using some glfw calls. But calling these after Terminate() results in some infinite loop. So they need to be manually destructed before then.